code/modules/mob/living/simple_mob/subtypes/vore/plants_ch/pitcher.dm

```diff
@@ -0,0 +1,321 @@
+	var/scent_strength = 5 //How much can a hungry pitcher confuse nearby people?
+	var/last_lifechecks = 0 //Timing variable to limit vore/hungry proc calls
+	var/list/pitcher_plant_lure_messages = null
+
+
+
+
+/mob/living/simple_mob/vore/pitcher_plant //Putting vore variables separately because apparently that's tradition.
+	vore_bump_chance = 100 
+	vore_bump_emote = "slurps up" //Not really a good way to make the grammar work with a passive vore plant.
+	vore_active = 1
+	vore_icons = 1
+	vore_capacity = 1
+	vore_pounce_chance = 0 //Plants only eat people who stumble into them.
+	swallowTime = 3 //3 deciseconds. This is intended to be nearly instant, e.g. victim trips and falls in.
+	vore_ignores_undigestable = 0
+	vore_default_mode = DM_DIGEST
+
+/mob/living/simple_mob/vore/pitcher_plant/init_vore()
+	..()
+	var/obj/belly/B = vore_selected
+	B.desc	= "You leaned a little too close to the pitcher plant, stumbling over the lip and splashing into a puddle of liquid filling the bottom of the cramped pitcher. You squirm madly, righting yourself and scrabbling at the walls in vain as the slick surface offers no purchase. The dim light grows dark as the pitcher's cap lowers, silently sealing the exit. With a sinking feeling you realize you won't be able to push the exit open even if you could somehow climb that high, leaving you helplessly trapped in the slick, tingling fluid. ((You can't escape this mob without help but you may use OOC Escape if you wish.))"
+	B.digest_burn = 0.5
+	B.digest_brute = 0
+	B.vore_verb = "trip"
+	B.name = "pitcher"
+	B.mode_flags = DM_FLAG_THICKBELLY
+	B.wet_loop = 0 //As nice as the fancy internal sounds are this is a plant.
+	B.digestchance = 0
+	B.escapechance = 0
+	B.fancy_vore = 1
+	B.vore_sound = "Squish2"
+	B.release_sound = "Pred Escape"
+	B.contamination_color = "purple"
+	B.contamination_flavor = "Wet"
+//Why is it we have all these customizeable belly options which nobody ever alters for mobs?
+
+	B.emote_lists[DM_HOLD] = list(
+		"Slick fluid trickles over you, carrying threads of sweetness.",
+		"Everything is still, dark, and quiet. Your breaths echo quietly.",
+		"The surrounding air feels thick and humid.")
+
+	B.emote_lists[DM_DIGEST] = list(
+		"The slimy puddle stings faintly. It seems the plant has no need to quickly break down victims.",
+		"The humid air settles in your lungs, keeping each breath more labored than the last.",
+		"Fluid drips onto you, burning faintly as your body heat warms it."
+		)
+
+	B.emote_lists[DM_DRAIN] = list(
+		"Each bead of slick fluid running down your body leaves you feeling weaker.",
+		"It's cramped and dark, the air thick and heavy. Your limbs feel like lead.",
+		"Strength drains from your frame. The cramped chamber feels easier to settle into with each passing moment.")
+
+/mob/living/simple_mob/vore/pitcher_plant/Life()
+	. = ..()
+	if(!.)
+		return
+
+	if(meat >= meatspeed) //If Life procs every 2 seconds that means it takes 20 seconds to digest a steak
```

Comment:
```suggestion
	var/lastmeat = meat //If Life procs every 2 seconds that means it takes 20 seconds to digest a steak
	meat = max(0,meat-meatspeed) //Clamp it to zero
	adjust_nutrition(lastmeat-meat) //If there's no meat, this will just be zero.
```
procs are generally more stable than editing values directly, and I imagine adjust_nutrition will probably clamp it to make the value not go too high. This change makes it so that the plant can actually finish digesting the meat, rather than leaving unfinished bits once it gets below a certain value. Will repeat similar things below.